HOWTO-make-a-release.md

```diff
@@ -0,0 +1,348 @@
+as PRs through Github.  However, if undisclosed security fixes are being
+applied, some other more private repository must be used.
+Worst case scenario, there have been times where the review process has
+been conducted by sending patches (made with `git format-patch`) by email
+and receiving approval as a reply.
+
+The release person and the reviewer must have a conversation to confirm
+or figure out how review shall be done.
+
+# Pre-publishing tasks
+
+All the actions in this section need to be repeated for each OpenSSL
+version released.
+
+## Freeze the source repository
+
+The day before the release, freeze the main repository.  This locks
+out everyone but the named user, who is doing the release, from doing
+any pushes.  Someone other than the person doing the release should
+run the command.  For example:
+
+    ssh openssl-git@git.openssl.org freeze openssl NAME
+
+## Prepare your repository checkouts
+
+You will need to checkout at least three working trees:
+
+-   one for the website
+
+        git clone openssl-git@git.openssl.org:openssl-web.git website
+
+-   one for extra tools
+
+        git clone openssl-git@git.openssl.org:openssl-tools.git tools
+
+-   At least one for the release to be done.  You may have to repeat
+    the release tasks that follow more than once, so it's often easier
+    to have separate copies for each release branch:
+
+        git clone openssl-git@git.openssl.org:openssl.git rel-111
+        cd rel-111
+        git branch --track OpenSSL_1_1_1-stable origin/OpenSSL_1_1_1-stable
+        git checkout OpenSSL_1_1_1-stable
+
+## Make sure that the source is up to date
+
+The person doing the release and the reviewer should both sanity-check
+the source to be released at this point.  Checks to consider includes
+building and verify that make test passes on multiple plaforms - Linux,
+Windows, etc.
+
+For each source checkout, make sure that the CHANGES.md / CHANGES and
+NEWS.md / NEWS files have been updated and reviewed.
+
+NEWS.md / NEWS should contain a summary of any changes for the release,
+and for a security release is (often just a list of the CVEs addressed.
+You should also update NEWS in the master branch to include details of
+all releases.  Just update the NEWS bullet points - do not change the
+release date, keep it as **under development**.
+
+Add any security fixes to the tree.  Commit them but *do not push to any
+public repository*.
+
+Make sure that the copyrights are updated.  This script will update
+the copyright markers and commit the changes (where $TOOLS stands for
+the openssl-tools.git checkout directory):
+
+    $TOOLS/release-tools/do-copyright-year
+
+Obtain approval for these commits from the reviewer and add the
+reviewed-by headers as required.
+If you have added security fixes, you must seek approval through
+extraordinary means as agreed with the reviewer.  This may involve
+pushing to a non-public repository you both have access to, or emailing
+patches, or ...
```

Comment:
typically using the private github fork

Reply:
Yeah, I've rewritten this bit to mention Github's security advisory mechanism

Reply:
Oh, and changed this section to have the same *Do not* and *Do* as other similar sections